pkg/ccl/changefeedccl/checkpoint/checkpoint_test.go

```diff
@@ -164,6 +166,166 @@ func TestCheckpointMake(t *testing.T) {
 	}
 }
 
+func TestCheckpointRestore(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ts := func(wt int64) hlc.Timestamp {
+		return hlc.Timestamp{WallTime: wt}
+	}
+
+	for name, tc := range map[string]struct {
+		initialHighWater       hlc.Timestamp
+		oldCheckpointTs        hlc.Timestamp
+		oldCheckpointSpans     []roachpb.Span
+		newCheckpoint          jobspb.TimestampSpansMap
+		expectedError          error
+		expectedFrontierForOld checkpointSpans
+		expectedFrontierForNew checkpointSpans
+	}{
+		"old checkpoint ts is empty": {
+			initialHighWater: ts(0),
+			oldCheckpointTs:  ts(0),
+			oldCheckpointSpans: []roachpb.Span{
+				{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+			},
+			newCheckpoint: jobspb.TimestampSpansMap{
```

Comment:
Could you use `jobspb.NewTimestampSpansMap` instead of manually creating a protobuf?

Reply:
Done.

Reply:
I think you missed one heh.

Reply:
Oops, done.